### ISSUE TITLE
`triggerRef` does not trigger `watch`

### ISSUE THREAD

**Issue by undergroundwires** [vuejs/vue#13117]
## Expected behavior

Watching values of manually triggered references should execute the watch callback.

## Reproduce

This test fails:

```javascript
import { nextTick } from 'vue'
test('triggerRef with watch', async () => {
    const a = shallowRef(1)
    let aTriggered = false
    watch(() => a.value, () => {
      aTriggered = true
    })
    triggerRef(a)

    await nextTick()  

    expect(aTriggered).toBe(true)
  })
```

## More information

Watching the `ref` itself works fine. According to post by @LinusBorg in https://github.com/vuejs/core/issues/1209#issuecomment-648953517, this is the expected design of Vue.

However, getting `value` of the `ref` does not work.  So `watch(() => a.value, ...)` won't work but, `watch(a, ...)` work. The behavior should be the same and consistent for `shallowRef`s where state tracking is disabled

`watchEffect` works fine for these `.value` calls. This is another undocumented inconsistency with `watch`.

Official Vue docs state nothing about caching for `watch`.

**Comment by posva** [comment]
As noted in the following comments, the current behavior is expected: https://github.com/vuejs/core/issues/1209#issuecomment-648976326

**Comment by LinusBorg** [comment]
@posva OP also opened this issue in the core repo, and there I noted that there is one weird thing: `watch(aShallowRef, ...)` *does* trigger the watch. Normal refers, or getter functions, don't. So *only* when you watch a shallowRef directly, it will be triggered.

This seeems to be by design, but I question the wisdom of that design. See https://github.com/vuejs/core/issues/9579#issuecomment-1805300580

Should we consider re-opening this issue here until we settled the raised question over in core?

**Comment by undergroundwires** [comment]
Hi, sorry for the duplication, I got confused and opened two issues.

I'd appreciate at least there would be some documentation regarding this. I've updated my application to use `watch(aShallowRef` to get this working for now.

A trivia: All of the language models including GPT 4.0 is confused about this behavior and expects `watch(() => aShallowRef.value)` to be triggered, not only humans 🙂
